lib/ansible/plugins/callback/grafana_annotations.py

```diff
@@ -0,0 +1,203 @@
+        env:
+          - name: GRAFANA_SECURE
+        default: 0
+      token:
+        description: Grafana API Token, allowing to authenticate when posting on the HTTP API.
+                     If not provided, the callback will be disabled.
+        env:
+          - name: GRAFANA_API_TOKEN
+'''
+
+import os
+import json
+import socket
+import getpass
+from datetime import datetime
+
+try:
+    import httplib
+except ImportError:
+    # Python 3
+    import http.client as httplib
+
+from ansible.plugins.callback import CallbackBase
+
+
+PLAYBOOK_START_TXT = """\
+Started playbook {playbook}
+
+From '{hostname}'
+By user '{username}'
+"""
+
+PLAYBOOK_ERROR_TXT = """\
+Playbook {playbook} Failure !
+
+From '{hostname}'
+By user '{username}'
+
+'{task}' failed on {host}
+
+debug: {result}
+"""
+
+PLAYBOOK_STATS_TXT = """\
+Playbook {playbook}
+Duration: {duration}
+Status: {status}
+
+From '{hostname}'
+By user '{username}'
+
+Result:
+{summary}
+"""
+
+
+def to_millis(dt):
+    return int(dt.strftime('%s')) * 1000
+
+
+class CallbackModule(CallbackBase):
+    """
+    ansible grafana callback plugin
+    ansible.cfg:
+        callback_plugins   = <path_to_callback_plugins_folder>
+        callback_whitelist = grafana
+    and put the plugin in <path_to_callback_plugins_folder>
+
+    This plugin makes use of the following environment variables:
+        GRAFANA_SERVER   (optional): defaults to localhost
+        GRAFANA_PORT     (optional): defaults to 3000
+        GRAFANA_SECURE   (optional): Set to 1 if HTTPs is implemented on Grafana.
+                                     defaults to 0 (false)
+        GRAFANA_API_TOKEN          : Grafana API authentication token
+    """
+
+    CALLBACK_VERSION = 1.0
+    CALLBACK_TYPE = 'aggregate'
+    CALLBACK_NAME = 'grafana'
+    CALLBACK_NEEDS_WHITELIST = True
+
+    def __init__(self):
+        super(CallbackModule, self).__init__()
+
+        token = os.getenv('GRAFANA_API_TOKEN')
+        self.secure = int(os.getenv('GRAFANA_SECURE', 0))
+        if self.secure not in [0, 1]:
+            self.secure = 0
+        if token is None:
+            self.disabled = True
+            self._display.warning("GRAFANA_API_TOKEN not defined in the environment")
+        self.grafana_server = os.getenv('GRAFANA_SERVER', 'localhost')
+        self.grafana_port = int(os.getenv('GRAFANA_PORT', 3000))
+        self.headers = {'Content-Type': 'application/json',
+                        'Authorization': 'Bearer %s' % token}
+        self.errors = 0
+        self.hostname = socket.gethostname()
+        self.username = getpass.getuser()
+        self.start_time = datetime.now()
+
+    def v2_playbook_on_start(self, playbook):
+        self.playbook = playbook._file_name
+        text = PLAYBOOK_START_TXT.format(playbook=self.playbook,
+                                         hostname=self.hostname, username=self.username)
+        data = {
+            'time': to_millis(self.start_time),
```

Comment:
`dashboardId` parameter is missing.
How do you specify on which dashboard annotations are created ?

Reply:
I did not put it on purpose because I felt that the annotations needs to be globals to allow one to display them on any dashboard.
But I can still add the feature allowing to provide a dashboard id.

Reply:
and I'll add panelId too.